sdk/scheduler/src/main/java/com/mesosphere/sdk/framework/OfferProcessor.java

```diff
@@ -207,12 +231,18 @@ private void processQueuedOffers(Duration queueWait) {
             // Match offers with work (call into implementation)
             final Timer.Context context = Metrics.getProcessOffersDurationTimer();
             try {
-                if (isActive()) {
+                if (checkStatus()) {
                     evaluateOffers(offers);
+                } else {
+                    // Offers not needed, at least not right now. Decline long.
+                    declineLong(offers);
                 }
             } finally {
                 context.stop();
             }
+
+            // After the status check, see if a revive is now needed:
+            reviveManager.reviveIfRequested();
```

Comment:
My only gripe with using the timeout of `OfferQueue.takeAll(...)` to turn the crank in the suppressed case is that we will be bounded by the offer wait time (5 seconds) with respect to recovery of failed tasks.

I can't immediately think of a better way to do things. Ideally, the `RecoveryManager(s)` would have a fast-lane to triggering a revive in the suppressed case.

Reply:
Yeah, it's almost certainly safer to keep the logic the same for all cases, but that 5 seconds is annoying.

Really, we just need them to change suppress to not include reservations.

Reply:
I mean, to be clear, there should only be a delay here if the revive is throttled. Otherwise the revive is sent out right as the current offer cycle finishes processing.

Reply:
@greggomann @gkleiman this is where having suppress _not_ suppress reservations would be <3